src/v3/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  Autocomplete as MuiAutocomplete,
+  AutocompleteRenderInputParams,
+  Box,
+  FormHelperText,
+  InputLabel,
+  TextField,
+} from '@mui/material';
+import { IdxOption } from '@okta/okta-auth-js/lib/idx/types/idx-js';
+import { h } from 'preact';
+
+import { getMessage } from '../../../../v2/ion/i18nTransformer';
+import { useAutoFocus, useOnChange, useValue } from '../../hooks';
+import {
+  FieldElement, UISchemaElementComponent,
+} from '../../types';
+
+const Autocomplete: UISchemaElementComponent<{
+  uischema: FieldElement
+}> = ({ uischema }) => {
+  const value = useValue(uischema);
+  const onChangeHandler = useOnChange(uischema);
+  const { label, focus } = uischema;
+  const {
+    inputMeta: {
+      // @ts-ignore expose type from auth-js
+      messages = {},
+      name,
+      options,
+    },
+    customOptions,
+  } = uischema.options;
+  const autocompleteProps = {
+    options: (customOptions ?? options)!,
+    getOptionLabel: (opt: IdxOption) => opt.label,
+    autoHighlight: true,
+    autoSelect: true,
+    disableClearable: true,
+    disablePortal: true,
+  };
+  const selectedOption = autocompleteProps.options.find((opt) => opt.value === value) || null;
+  const error = messages?.value?.[0] && getMessage(messages.value[0]);
+  const focusRef = useAutoFocus<HTMLSelectElement>(focus);
```

Comment:
When this element takes autofocus does it cause the dropdown to open?

Reply:
It does not because we did not enable `openOnFocus` in MUI's Autocomplete component. 
(it is false by default)

See: https://mui.com/material-ui/api/autocomplete/
```
openOnFocus | bool | false | If true, the popup will open on input focus.
```